core/commands/resolve.go

```diff
@@ -78,6 +84,16 @@ Resolve the value of another name recursively:
 			depth = namesys.DefaultDepthLimit
 		}
 
+		if strings.HasPrefix(name, "/ipfs/") || !strings.HasPrefix(name, "/") {
+			resolved, err := resolveIpfsPath(req.Context(), n.Resolver, name)
+			if err != nil {
+				res.SetError(err, cmds.ErrNormal)
+				return
+			}
+			res.SetOutput(&ResolvedPath{resolved})
+			return
+		}
```

Comment:
I want:

```
> ipfs resolve /ipns/NAME
/ipfs/HASH_OF_BAR
```
your PR gives
```
> ipfs resolve /ipns/NAME
/ipfs/HASH/foo/bar
```

> Maybe it would make sense to do the former by default, and the latter when the -r flag is used?

No, we should match all the other Resolve interfaces: resolve should always resolve recursively, all the way--  unless passing in a constant number as the "number of times to resolve" (then it runs until the number runs out). Users could pass `-n=1` to resolve only once. there are long discussions with wking on the issues list if you're curious. 

there's no need to implement `-n=1` support now. just make it resolve all the way to the end.

users can get the raw value of a name with `ipfs name resolve`